Reduce Square client timeout but retry

Syfaro opened the pull request:
This is completely untested, but apparently we can very easily have the Square API client retry requests. I'm also not sure if 2 seconds is long enough or if 5 retries is too many.

rechner commented on registration/payments.py:
```diff
@@ -10,12 +10,13 @@
 from .models import *
 
 client = Client(
+    timeout=2,
```
2s seems rather quick, especially for a payments API where we're waiting for completion instead of just capture/auth first.  I'm thinking more in the range of 5 - 10s at least, though maybe 15s is a more appropriate lower bound.

It'd be good to try and instrument external API calls with Prometheus. Usually there's drop in libraries or wrappers for `requests` or similar client libraries but we could also just instrument a histogram with our payments code as well to get better visibility on these timings.

rechner commented:
also of note: combined timeout times retries cannot exceed our server body reply timeout, which is set to the default by nginx to 60s.

Syfaro commented:
I've updated the retry logic to a 10s timeout with up to 5 retries, which gives us ~10 seconds to play with before hitting our nginx reply timeout. This is still definitely too long for certain types of requests, so in the future I might create multiple clients for different categories of requests. I also added instrumentation to all of our requests to the Square API to see how long each type of request takes.

rechner reviewed (approved):
📈 📊
